### PR TITLE
Kvcache changes to support deepseek MLA

### PR DESCRIPTION
**Pull request by mitalisi**
# Description
Changes to support caching for deepseek mla
b/402537258

*Notice 1:* Once all tests pass, the "pull ready" label will automatically be assigned.
This label is used for administrative purposes. Please do not add it manually.

*Notice 2:* For external contributions, our settings currently require an approval from a MaxText maintainer to trigger CI tests.

# Tests

Please describe how you tested this change, and include any instructions and/or
commands to reproduce.

# Checklist

Before submitting this PR, please make sure (put X in square brackets):
- [ ] I have performed a self-review of my code.
- [ ] I have necessary comments in my code, particularly in hard-to-understand areas.
- [ ] I have run end-to-end tests tests and provided workload links above if applicable.
- [ ] I have made or will make corresponding changes to the doc if needed.

### CODE REVIEW

**Review by gagika — COMMENTED** [review]
Thanks for the feature!

Can we also have a correctness test with Naive MLA implementation?

e.g. prefill/decode with MLA compressed should produce the same logits (up to a tolerance) with naive implementation. It can be a follow up PR

**Review by gagika — APPROVED** [review]
Thanks!

**Review by vipannalla — APPROVED** [review]
looks good

File: MaxText/inference/kvcache.py [hunk 1]
```diff
@@ -354,17 +361,17 @@ def chunked_prefill_kv_cache(self, key: Array, value: Array, decoder_segment_ids
     function updates current key and value at the desired location and returns entire key and value
 
     """
-    batch, _, heads, key_head_size = key.shape
-    batch, _, heads, value_head_size = value.shape
```
[the remark this answers is not in the document]
Comment (mitalisi) [reply MaxText/inference/kvcache.py:358]: > How did this the current functionality work for models using GQA/MQA? since the code used value_heads for key too?

That would have different number of query and kv heads but same number of key and value heads.
